src/foam/util/Arrays.java

```diff
@@ -172,4 +170,18 @@ protected static void sortRange(Object[] a, Comparator c, int skip, int limit, i
     sortRange(a, c, skip, limit, p2 + 1, end);
   }
 
+  public static Object[] toArray(Object o) {
+    if ( o.getClass().isArray() ) {
```

Comment:
This is going to be very expensive for a comparison because it is going to copy all of the data. Even if the arrays aren't even the same length, we're still going to copy them both. Why do we need this? If we do, better to just write 10 efficient compare array methods than doing this.